src/status_im2/contexts/chat/composer/link_preview/events_test.cljs

```diff
@@ -0,0 +1,212 @@
+(ns status-im2.contexts.chat.composer.link-preview.events-test
+  (:require [status-im2.contexts.chat.composer.link-preview.events :as events]
+            [cljs.test :refer [is deftest testing]]))
+
+(def url-github "https://github.com")
+(def url-gitlab "https://gitlab.com")
+(def preview-github {:url url-github :thumbnail nil})
+(def preview-gitlab {:url url-gitlab :thumbnail nil})
+(def request-id "fb973113-86f3-49c0-b750-5734a2e77a90")
+
+(deftest unfurl-urls
+  (testing "clear up state when text is empty"
+    (let [cofx                       {:db {:chat/link-previews {:unfurled   {}
+                                                                :cache      {}
+                                                                :request-id "123"}}}
+          {:keys [db json-rpc/call]} (events/unfurl-urls cofx " ")]
+      (is (= {:chat/link-previews {:cache {}}}
+             db))
+      (is (nil? call))))
+
```

Comment:
Nice work, @ilmotta! I'll add some feedback on the tests here, which you may like or not based on your preferences, feel free to ignore if you don't agree with them:

- I think the `{:keys [db json-rpc/call]}` is a bit confusing to someone reading the code. It may be that you want to emphasize the existence of the two code paths in the `events/unfurl-url` fn, but my preference is to just see data go in and data coming out, like this:
``` clojure
(is (= (events/unfurl-urls
          {:db {:chat/link-previews {:unfurled   {}
                                     :cache      {}
                                     :request-id "123"}}} " ")
         {:db {:chat/link-previews {:cache {}}}}))
```

Reply:
Good point, in this example I really don't need to destructure at all.